lib/plugins/aws/provider/awsProvider.js

```diff
@@ -222,6 +222,77 @@ class AwsProvider {
             required: ['Fn::Sub'],
             additionalProperties: false,
           },
+          awsKmsArn: {
```

Comment:
I think we should introduce it directly here: https://github.com/serverless/serverless/blob/29d5fe75104c9bb0f44531e1c4b3491d4051f6b6/lib/configSchema.js#L114 because of https://github.com/serverless/serverless/blob/29d5fe75104c9bb0f44531e1c4b3491d4051f6b6/lib/configSchema.js#L12

I've opened issue to recover from that wierd lone AWS setting in `service` context https://github.com/serverless/serverless/issues/8261  still we can't remove it until next major

Reply:
@medikoo in order to mutualize definition at `service` and `functions` level, I had to modify a few things :

- first, using `$ref` within one of `service` property definition requires any `service` properties to be writable (since `normalizeSchemaObject` will replace those `$ref` with actual reference definition). I therefor replace existing deepFreeze of `service` object by a simple freeze of `service.name` property, keeping `service.awsKmsKeyArn` writable by `normalizeSchemaObject` (see https://github.com/serverless/serverless/blob/d401dda71e09627609b67aae1ce1a7be78c53a58/lib/classes/ConfigSchemaHandler/index.js#L31:L32)
- second, I replace unnecessarily overcomplicated pattern `'^arn:(aws[a-zA-Z-]*)?:kms:[a-z0-9-]+-\\d+:\\d{12}:[^\\s]+$'` with `'^arn:aws[a-z-]*:kms'`

Are you fine with both those changes ?

Reply:
@fredericbarthelet yes, that looks great.

Now, I wonder whether there's value in those `Object.freeze`. It was initially introduced as we thought that we will expose schema as whole object to plugins for extensions, so freezing some core parts felt as right precautionary thing to do, but after all, we ended with different extension API.

I'd consider removing this `Object.freeze` wraps (but ofc that's not in scope of this PR)